lib/stringMsg.js

```diff
@@ -98,18 +102,37 @@ exports.parsePCDIN = (input) => {
     { coalesced: true, prefix, timer, timestamp: new Date(timer) },
   )
 }
-exports.parseN2KOver0183 = exports.parsePCDIN
-
-function toPaddedHexString(num, len) {
-  str = num.toString(16).toUpperCase();
-  return "0".repeat(len - str.length) + str;
-}
 
 exports.encodePCDIN = ({ prefix = '$PCDIN', pgn, data, dst = 255}) => {
   const sentence = [ prefix, toPaddedHexString(pgn, 6), '0000180C', hexByte(dst).toUpperCase(), byteString(data, '').toUpperCase()].join(',')
   return sentence + compute0183Checksum(sentence)
 }
 
+// $MXPGN,01F801,2801,C1308AC40C5DE343*19
+exports.isMXPGN = startsWith('$MXPGN,')
+exports.parseMXPGN = (input) => {
+  const [ prefix, pgn, attr_word, data ] = input.split(',')
+
+  const send_prio_len = (parseInt(attr_word.substr(0,2), 16).toString(2)).padStart(8, '0');
+  const addr = (parseInt(attr_word.substr(2,2), 16));
+  const send = parseInt(send_prio_len.substr(0,1), 2);
+  const prio = parseInt(send_prio_len.substr(1,3), 2);
+  const len = parseInt(send_prio_len.substr(4,4), 2);
+  let src, dst;
+  send ? dst = addr: src = addr;
+  
+  return buildMsg(
+    buildCanId(0, parseInt(pgn, 16), 255, parseInt(src, 16)),
+    'MXPGN',
+    Buffer.from(rmChecksum(data), 'hex'),
+    { coalesced: true, prefix },
+  )
+}
+
+exports.encodeMXPGN = ({ prefix = '$MXPGN', pgn, data, dst = 255}) => {
```

Comment:
This function as it currently is does not work on my Miniplex because the attribute word is not created correctly. Here is a suggested change for this function. I have tested it and it works well with my Miniplex. I am happy to fork and submit a pull request if you prefer.

```
exports.encodeMXPGN = ({ prefix = '$MXPGN', pgn, prio, src, data }) => {
  if (src > 255) src = 255;
  if (!prio) prio = 3;
  if (!src) src = 255;
  const dataLength = hexByte(128 + (prio * 16) + (byteString(data, '').toUpperCase().length / 2)).toUpperCase()
    const attribWord = dataLength + hexByte(src).toUpperCase()

    var buff = Buffer.from(byteString(data, ''), 'Hex');
    for (var i = 0, j = buff.length - 1; i < j; ++i, --j) {
        var t = buff[j]

        buff[j] = buff[i]
        buff[i] = t
    }
    //console.log(buff.toString('Hex').toUpperCase());

    const sentence = [prefix, toPaddedHexString(pgn, 6), attribWord, buff.toString('Hex').toUpperCase()].join(',')
  return sentence + compute0183Checksum(sentence)
}
```

Reply:
Yes, please do a PR. Thanks

Reply:
Thanks so much for catching this.